base/reflection.jl

```diff
@@ -370,12 +400,12 @@ function code_typed_by_type(@nospecialize(tt::Type);
         throw(ArgumentError("'debuginfo' must be either :source or :none"))
     end
     tt = to_tuple_type(tt)
-    matches = Compiler.findall(tt, Compiler.method_table(interp))
+    matches = invoke_interp_compiler(passed_interp, :_findall_matches, interp, tt)
```

Comment:
What’s the reason for passing `passed_interp` here? If we pass `interp` always here, we could remove the `interp === nothing` branch in `invoke_interp_compiler`, which would make the code flow clearer.

Reply:
I wanted to account for the possibility that somebody may pass a Core.Compiler associated interpreter, but wants to run it in the current world age. I'm not fully happy with it - I think we may change up the way that we register the typeinf function a bit, but I think it's fine for now.